### ISSUE TITLE
Bug in v0.14.1 expire_escrow_ratification

### ISSUE THREAD

**Issue by abitmore** [steemit/steem#397]
An escrow should expire and be cancelled automatically if either `to` or `agent` hasn't approved it before `ratification_deadline`?

Currently the index `expire_escrow_ratification` is ordered by `tuple <to_approved, agent_approved, ratification_deadline>`.

When an escrow is approved by `to` or `agent` but not both, it will fall behind other escrows which have nobody approved, so it will not be processed when met `ratification_deadline` (when "the others" haven't expired yet), which means if a sender accidentally set a wrong agent and approved by receiver, she will have no chance to get back the fund if there is another "bad" escrow (approved by nobody) exists (in front).

And worse, with current code, once an escrow is approved by `to` or `agent`, it will never be processed in `expire_escrow_ratification`. ([code](https://github.com/steemit/steem/blob/v0.14.1/libraries/chain/database.cpp#L2419))
```
   while( escrow_itr != escrow_idx.end() && !escrow_itr->to_approved 
      && !escrow_itr->agent_approved && escrow_itr->ratification_deadline <= head_block_time() )
{
   ...
}
```

Anyway, sender should have a way to cancel an escrow when receiver and/or agent haven't approved yet, but perhaps this belongs to another ticket.

**Comment by mvandeberg** [comment]
Good catch.

In regards to the sender being able to cancel. The current implementation requiring ratification is in place of multisig. If we used multisig and the sender signed the transaction and forwarded it on to `to` and `agent` they have no way of preventing them from signing and broadcasting the transaction other than to plead with them. Because of this, I don't think that ability is necessary. Instead, I recommend that escrow has a relatively short ratification deadline (24 to 48 hours). The specifics of the escrow should have already been discussed off chain and ratification is only the formalization of the contract on chain. It should be a quick process.
